x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -108,8 +108,21 @@ timeSeriesCommand
     : TS indexPatternAndMetadataFields
     ;
 
-indexPatternAndMetadataFields:
-    indexPattern (COMMA indexPattern)* metadata?
+indexPatternAndMetadataFields
+    : indexPatternOrSubquery (COMMA indexPatternOrSubquery)* metadata?
+    ;
+
+indexPatternOrSubquery
+    : indexPattern
+    | {this.isDevVersion()}? subquery
+    ;
+
+subquery
+    : LP fromCommand (PIPE subqueryProcessingCommand)* RP
+    ;
+
+subqueryProcessingCommand
```

Comment:
Do we need this explicit parser rule, given we don't rule out any of the processing commands? Can stay, I was just wondering, since I see the same pattern with fork.